### ISSUE TITLE
Latest update (0.2.23) broke select filtering

### ISSUE THREAD

**Issue by dsl101** [koalyptus/TableFilter#209]
I just noticed that since updating (to try to get the thousand separator filtering functionality), the select filter columns now _only_ show 'Clear'. The other options are missing, which wasn't the case with the previous version. Did something else change in the code which might affect that?

**Comment by dsl101** [comment]
Sorry - scrap that. There was a discrepancy between the table header and content which I _think_ put the columns out of alignment by 1. All working again.

**Comment by dsl101** [comment]
Thought the underlying cause might be interesting - I know it's my fault, as presumably the table wasn't full spec compliant. But I can't imagine it's the only one :)

I have an 11-column table of data, and wanted to put a 'delete row' icon as the 12th column. But I didn't want it to have a header - so the delete icons would kind of hang off the right of the table. I set this up with 11 `<th>` elements, but then added 12 `<td>` elements to each subsequent row. In the browser this looks fine - but it caused tablefilter to break badly. Filtering didn't work at all, and any columns marked as 'Select' got a drop box which only showed the 'All' option - they weren't populated with the rest of the data.

I guess it would be nice if it could cope gracefully with this scenario, but I know it's probably not in spec to have different numbers of columns to header elements. Once I added a blank header entry, all was well, and it doesn't look too bad, so I'm happy with that.
